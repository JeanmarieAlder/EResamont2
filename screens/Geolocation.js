import React, { useState, useEffect } from "react";
import { View, Text, StyleSheet } from "react-native";
import { globalStyles } from "../styles/global";
import { WebView } from "react-native-webview";

import * as Location from "expo-location";
import * as Permissions from "expo-permissions";
import ButtonView from "../components/ButtonView";

export default function Geolocation({ navigation }) {
  let [location, setLocation] = useState(null);

  useEffect(() => {
    getGeolocationAsync();
  }, []);

  let getGeolocationAsync = async () => {
    let { status } = await Permissions.askAsync(Permissions.LOCATION);
    if (status !== "granted") {
      this.setState({
        errorMessage: "Permission to access location was denied"
      });
    }

    let location = await Location.getCurrentPositionAsync({});
    setLocation(location);
  };

  return (
    <View style={{ flex: 1 }}>
      <ButtonView
        value={"Get location"}
        style={globalStyles.midataButton}
        onClick={() => getGeolocationAsync()}
      ></ButtonView>
      {location ? (
        <View style={localStyles.topContainer}>
          <Text>Latitude: {location.coords.latitude}</Text>
          <Text>Longitude: {location.coords.longitude}</Text>
          <Text>Altitude (estimation): {location.coords.altitude}</Text>
          <View style={globalStyles.topMenuDivider} />
          <View style={globalStyles.mapContainer}>
            <WebView
              source={{
                uri: `https://www.google.com/maps/place/${location.coords.latitude},%20${location.coords.longitude}`
              }}
            ></WebView>
          </View>
        </View>
      ) : (
        <View style={localStyles.loaderViewMain}>
          <Text>
            Allow this app to use your location by clicking on "Get location".
          </Text>
        </View>
      )}
    </View>
  );
}

const localStyles = StyleSheet.create({
  topContainer: {
    flex: 1
  }
});
