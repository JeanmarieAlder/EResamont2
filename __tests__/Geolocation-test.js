import * as React from "react";
import renderer from "react-test-renderer";
import { act } from "react-test-renderer";
import Geolocation from "../screens/Geolocation";
import * as Permissions from "expo-permissions";
import { ToastAndroid } from "react-native";
import { render, fireEvent } from "react-native-testing-library";

jest.mock("expo-permissions");
jest.mock("expo-location");
ToastAndroid.show = jest.fn();

describe("Geolocation screen", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it("should render correctly", async () => {
    Permissions.askAsync = jest.fn(() => {
      return { status: "granted" };
    });
    let component = render(<Geolocation></Geolocation>);

    expect(component).toMatchSnapshot();
  });
  it("should render correctly when permission denied", async () => {
    Permissions.askAsync = jest.fn(() => {
      return { status: "denied" };
    });
    let component;
    await act(async () => {
      component = renderer.create(<Geolocation></Geolocation>).toJSON();
    });

    expect(ToastAndroid.show).toHaveBeenCalledWith(
      "Permission denied",
      ToastAndroid.SHORT
    );
  });
});
