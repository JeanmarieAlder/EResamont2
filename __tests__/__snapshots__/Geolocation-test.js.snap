// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geolocation screen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Please turn on location permission
    </Text>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      (App info -&gt; Permission -&gt; Location)
    </Text>
  </View>
</View>
`;
